src/shogun/kernel/ANOVAKernel.h

```diff
@@ -105,32 +86,21 @@ class ANOVAKernel: public DotKernel
 	 */
 	float64_t compute_rec2(int32_t idx_a, int32_t idx_b);
 
-	/** Casts the given kernel to CANOVAKernel.
-	 * @param kernel Kernel to cast. Must be CANOVAKernel. Might be NULL
-	 * @return casted CANOVAKernel object, NULL if input was NULL
+	/** Casts the given kernel to ANOVAKernel.
+	 * @param kernel Kernel to cast. Must be ANOVAKernel. Might be NULL
+	 * @return casted ANOVAKernel object, NULL if input was NULL
 	 */
 	static std::shared_ptr<ANOVAKernel> obtain_from_generic(const std::shared_ptr<Kernel>& kernel);
-protected:
-
-	/**
-	 * compute kernel for specific feature vectors
-	 * corresponding to [idx_a] of left-side and [idx_b] of right-side
-	 * @param idx_a left-side index
-	 * @param idx_b right-side index
-	 * @return kernel value
-	 */
-	virtual float64_t compute(int32_t idx_a, int32_t idx_b);
-
-	/** register params */
-	void register_params();
 
 private:
 	float64_t compute_recursive1(float64_t* avec, float64_t* bvec, int32_t len);
 	float64_t compute_recursive2(float64_t* avec, float64_t* bvec, int32_t len);
 
 protected:
+	virtual float64_t compute(int32_t idx_a, int32_t idx_b);
```

Comment:
minor, but might be worth leaving compute where it was? So that the class members are separate from the class functions?

Reply:
I see your point